rmw_opendds_cpp/src/type_support_common.hpp

```diff
@@ -55,6 +55,23 @@
     } \
   }
 
+inline const rosidl_message_type_support_t *
+rmw_get_message_type_support(const rosidl_message_type_support_t * type_support)
+{
+  if (!type_support) {
+    RMW_SET_ERROR_MSG("type support is null");
+    return nullptr;
+  }
+  const rosidl_message_type_support_t * ts =
+    //get_message_typesupport_handle(type_support, rosidl_typesupport_opendds_cpp::typesupport_identifier);
+    get_message_typesupport_handle(type_support, "rosidl_typesupport_c"); //for temporary testing
```

Comment:
How do we determine what this string (2nd argument of get_message_typesupport_handle) should be?

Reply:
In the original macro (see below), a rosidl_message_type_support_t is initialized by calling 
rmw_get_message_type_support first with rosidl_typesupport_opendds_c__identifier,
and then (if returned null) with rosidl_typesupport_opendds_cpp::typesupport_identifier.

These calls will end up in strcmp of these parameters with the passed-in 
rosidl_message_type_support_t->rosidl_message_typesupport_handle_function->typesupport_identifier.

The current value of that typesupport_identifier in my test runs is "rosidl_typesupport_c". 
It is not our opendds specific and could not match either our c or c++ typesupport_identifier.

At this point, I'm not sure how rmw framework passes that value here or 
whether it passes that value as default when no implementation specific value is provided, or
how an implementation should provide its typesupport_identifier, through configuration, file or code.

I took Jeremy's suggestion and made an inline function which made debugging easier.
I only did the cpp part for now and even that is commented out and a temp string is used to pass strcmp.
I'm going to run some other implementation code and hope to get some answers.


#define RMW_OPENDDS_EXTRACT_MESSAGE_TYPESUPPORT(TYPE_SUPPORTS, TYPE_SUPPORT, RET_VAL) \
  if (!TYPE_SUPPORTS) { \
    RMW_SET_ERROR_MSG("type supports handle is null"); \
    return RET_VAL; \
  } \
  const rosidl_message_type_support_t * TYPE_SUPPORT = \
    get_message_typesupport_handle( \
    TYPE_SUPPORTS, rosidl_typesupport_opendds_c__identifier); \
  if (!TYPE_SUPPORT) { \
    TYPE_SUPPORT = get_message_typesupport_handle( \
      TYPE_SUPPORTS, rosidl_typesupport_opendds_cpp::typesupport_identifier); \
    if (!TYPE_SUPPORT) { \
      char __msg[1024]; \
      snprintf( \
        __msg, 1024, \
        "type support handle implementation '%s' (%p) does not match valid type supports " \
        "('%s' (%p), '%s' (%p))", \
        TYPE_SUPPORTS->typesupport_identifier, \
        static_cast<const void *>(TYPE_SUPPORTS->typesupport_identifier), \
        rosidl_typesupport_opendds_cpp::typesupport_identifier, \
        static_cast<const void *>(rosidl_typesupport_opendds_cpp::typesupport_identifier), \
        rosidl_typesupport_opendds_c__identifier, \
        static_cast<const void *>(rosidl_typesupport_opendds_c__identifier)); \
      RMW_SET_ERROR_MSG(__msg); \
      return RET_VAL; \
    } \
  }

Reply:
GitHub erased all the \ character in the macro.

Reply:
@lij-oci I think we covered this. I think we can resolve this conversation since you converted to the inline function or am I mistaken?